### PR TITLE
Ability to configure neutrino useragent

### PR DESCRIPTION
**Pull request by mrfelton**
This PR adds the ability to alter the neutrino useragent string so that neutrino clients can identify themselves as they please.

It exposes the neutrino `UserAgentVersion` and `UserAgentName` config options which can be set by starting lnd with the `--neutrino.useragentversion=` and `--neutrino.useragentname=` flags.

This enables btcd nodes to better identify connecting clients. For example, it could be used in conjunction with [btcd's `agentblacklist` and `agentwhitelist` settings]( https://github.com/btcsuite/btcd/pull/1418) to ensure that only clients that identify themselves in a specific way are allowed to connect.

### CODE REVIEW

**Comment by Roasbeef** [comment]
So do we want to block on fixing that issue, or is this g2g?

**Comment by wpaulino** [comment]
> So do we want to block on fixing that issue, or is this g2g?

It should be a simple fix. I can pick it up if @mrfelton is unable to continue.

File: chainregistry.go [hunk 1]
```diff
@@ -785,6 +785,7 @@ func initNeutrinoBackend(cfg *Config, chainDir string) (*neutrino.ChainService,
 
 	neutrino.MaxPeers = 8
 	neutrino.BanDuration = time.Hour * 48
+	neutrino.UserAgentName = cfg.NeutrinoMode.UserAgentName
```
[the remark this answers is not in the document]
Comment (mrfelton) [reply chainregistry.go:788]: I've added a new commit that sets these default values.

**Review by cfromknecht — APPROVED** [review]
LGTM 🏅 thanks @mrfelton!

**Comment by cfromknecht** [comment]
@Roasbeef the primary comment has been addressed. longer term we should figure out a solution in the neutrino repo for modifying these values without overwriting globals

**Pull request merged**